projects/metis/src/app/dataset/reportsimple/reportsimple.component.html

```diff
@@ -1,102 +1,107 @@
+    {
+      label: 'Close',
+      cssClass: 'button-filled'
+    }
+  ]"
+>
+  <ng-content *ngTemplateOutlet="tmpReportContent"></ng-content>
+</lib-modal>
+
+<ng-template #tmpReportContent>
+  <app-load-animation
+    [resources]="{ reportErrors: false }"
+    *ngIf="reportLoading"
+  ></app-load-animation>
+  <app-notification
+    [notification]="notification"
+    (closed)="notification = undefined"
+  ></app-notification>
+  <div #contentRef>
+    <div
+      *ngIf="reportRequest.message"
+      [ngClass]="{ 'fail-message-separator': reportRequest.errors }"
+    >
+      {{ reportRequest.message }}
     </div>
-    <div class="content">
-      <app-load-animation
-        [resources]="{ reportErrors: false }"
-        *ngIf="reportLoading"
-      ></app-load-animation>
-      <app-notification
-        [notification]="notification"
-        (closed)="notification = undefined"
-      ></app-notification>
-      <div #contentRef>
-        <div
-          *ngIf="reportRequest.message"
-          [ngClass]="{ 'fail-message-separator': reportRequest.errors }"
-        >
-          {{ reportRequest.message }}
-        </div>
-        <div *ngIf="reportRequest.errors; let reportErrors">
-          <ul *ngFor="let r of reportKeys(reportErrors)">
-            <li *ngFor="let rk of reportKeys(reportErrors[r])">
-              <strong class="capitalise">{{ splitCamelCase(rk) }}:&nbsp;</strong>
-              <span *ngIf="!isObject(reportErrors[r][rk])">
-                <app-text-with-links [text]="reportErrors[r][rk]"></app-text-with-links>
-              </span>
-              <ul class="report-errors" *ngIf="isObject(reportErrors[r][rk])">
-                <ng-container *ngFor="let subrk of reportKeys(reportErrors[r][rk])">
-                  <ng-container *ngFor="let sub of reportKeys(reportErrors[r][rk][subrk])">
-                    <li class="omit-from-copy" *ngIf="sub === 'identifier' && isDownloadable()">
-                      <span>
-                        <strong class="capitalise">Record XML:&nbsp;</strong>
-                        <a
-                          class="btn-download"
-                          (click)="
-                            downloadRecord(
-                              reportErrors[r][rk][subrk][sub],
-                              reportErrors[r][rk][subrk]
-                            )
-                          "
-                        >
-                          <strong>Download</strong>
-                        </a>
-                      </span>
-                    </li>
-                    <li
-                      class="download-error omit-from-copy"
-                      *ngIf="
-                        sub === 'identifier' && reportErrors[r][rk][subrk].downloadError;
-                        let downloadError
+    <div *ngIf="reportRequest.errors; let reportErrors">
+      <ul *ngFor="let r of reportKeys(reportErrors)">
+        <li *ngFor="let rk of reportKeys(reportErrors[r])">
```

Comment:
What are these variable `r` and `rk` meant to be?

Reply:
The variables `r` (report) and `rk` (report key) were already in place and not introduced as part of this PR.

I wanted to do the minimum amount of refactoring to get all modal behaviour configured in the one place, which in the case of this file was about changing the wrapping element to a modal dialog instance.

But I wonder if having longer and explicit names would actually make the file more readable?  For me expressions can become less readable when they contain both the singular and the plural (`reportKey` / `reportKeys` etc.), especially in lines 41-56 of this file, which become far longer with the unabbreviated names (especially when `subrk` is introduced!)

Reply:
the names `report` and `reportKey` made sense in this context, me as a external reader of the code I was able to understand this part of the code.